src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -333,6 +341,22 @@ private function parseDefinition(\DOMElement $service, $file, array $defaults =
             $definition->addAutowiringType($type->textContent);
         }
 
+        $bindingNodes = $this->getChildren($service, 'binding');
+        if (empty($bindingNodes) && !empty($defaults['bindings'])) {
+            $bindingNodes = $defaults['bindings'];
+        }
+
+        $bindings = array();
+        foreach ($bindingNodes as $binding) {
+            if (!$binding->hasAttribute('id')) {
+                throw new InvalidArgumentException(sprintf('The "id" attribute for binding type "%s" for service "%s" in %s must be defined.', $type, (string) $service->getAttribute('id'), $file));
+            }
+
+            $bindings[$binding->getAttribute('type')] = new Reference($binding->getAttribute('id'));
```

Comment:
type should renamed to eg "key" (to allow "type" to be used as suggested previously)